core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -490,17 +534,18 @@ object CachedDataFrame
     if (dataLen == 0) return Iterator.empty
 
     val codec = CompressionCodec.createCodec(SparkEnv.get.conf)
-    val input = new Input(data, offset, dataLen)
```

Comment:
Input is at driver side. We have not tested driver side memory manager much. I will file a ticket and check if we need some more changes at driver side.